core/src/main/java/org/camunda/spin/spi/DataFormatProvider.java

```diff
@@ -32,4 +34,12 @@
    */
   DataFormat<?> createInstance();
 
+  /**
+   * Adds a Map of configuration properties to the DataFormatProvider implementation to be
+   * used to configure the provided data format.
+   */
+  default void setConfigurationProperties(Map<String, Object> configurationProperties) {
+    // don't use configuration properties by default
+  };
```

Comment:
Is it nicer if we have a second `createInstance` method that accepts properties? With the current solution, the user has to make assumptions about in which order the methods are called, and in addition the provider becomes a stateful object.

Reply:
A second `createInstance` method was my initial solution. I went with the `setter` instead since I thought that users would prefer a separate method to extract the necessary properties.

However, your point about the order of execution, as well as having a stateful object in Spin (which is pretty stateless) is more valid. I'll refactor the code.